modules/afsocket/afinet-dest.c

```diff
@@ -152,6 +189,23 @@ _determine_port(const AFInetDestDriver *self)
   return port;
 }
 
+static void
+_setup_next_hostname(AFInetDestDriver *self)
+{
+  gchar *next_dst_candidate = _get_next_destination_candidate(self);
+  if (strcmp(next_dst_candidate, self->hostname) == 0)
+    {
+      msg_warning("The server is inaccessible, trying the old host again", evt_tag_str("host", self->hostname));
```

Comment:
Can we unconditionally write warning here? Maybe I miss something, but because this function is called in afinet_dd_setup_addresses() unconditionally: in the very first attempt, of when there are no candidates at all, even when there was no error, we hit this line. In those cases this warning is false positive.

Reply:
I don't understand this message. Does it get logged when we tried all servers in our list? Because if that's the case the log message should be clarified.

Reply:
@bazsi: you are the original author of the log message :)
But yes, agree, list handling and other log messages, could be clarified. This is just the very-first stage of the port (adopt the functionality from pe and let's make it work). After this step it is possible to extend zts testcases by failover tests. My plan is that after all the tests passed (and have a green set of testcases) I'll do the refactors.

Reply:
I am absolutely positive that I didn't write that code myself. I may have ported it though :)

failover-servers was done either by @juhaszviktor or Attila Nagy, can't remember exactly.

Reply:
:)